### PR TITLE
[dotnet] Add firefox capabilities

### PR DESCRIPTION
**Pull request by nadvolod**

### CODE REVIEW

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- dotnet/SimpleSauce.Test/Browsers/SafariTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/FirefoxTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/ChromeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/EdgeTests.cs  1
         

Complexity decreasing per file
==============================
+ dotnet/SimpleSauce.Test/SauceOptionsTests.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308&bid=15864760)

File: dotnet/Simple.Sauce/SauceOptions.cs [hunk 1]
```diff
@@ -102,5 +104,23 @@ public string SetCorrectPlatformVersion(string safariBrowserVersion)
...
+            ConfiguredFirefoxOptions = new FirefoxOptions()
+            {
+                BrowserVersion = DefaultBrowserVersion,
+                PlatformName = DefaultPlatform
+            };
+        }
+
+        public void WithFirefox(string version)
+        {
+            ConfiguredFirefoxOptions = new FirefoxOptions()
```
Comment (nadvolod) [inline dotnet/Simple.Sauce/SauceOptions.cs]: Codacy Issue found: [Remove these redundant parentheses.](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- dotnet/SimpleSauce.Test/Browsers/ChromeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/EdgeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/FirefoxTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/SafariTests.cs  1
         

Complexity decreasing per file
==============================
+ dotnet/SimpleSauce.Test/SauceOptionsTests.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308&bid=15864760)

File: dotnet/Simple.Sauce/SauceOptions.cs (hunk 1 above)
Comment (nadvolod) [inline dotnet/Simple.Sauce/SauceOptions.cs]: Codacy Issue found: [Remove these redundant parentheses.](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308)

**Review by titusfortner — APPROVED** [review]
lgtm

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dotnet/SimpleSauce.Test/Browsers/SafariTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/ChromeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/FirefoxTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/EdgeTests.cs  1
         

Complexity decreasing per file
==============================
+ dotnet/SimpleSauce.Test/SauceOptionsTests.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308&bid=15864760)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dotnet/SimpleSauce.Test/Browsers/SafariTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/EdgeTests.cs  1
- dotnet/Simple.Sauce/SauceSession.cs  1
- dotnet/SimpleSauce.Test/Browsers/ChromeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/FirefoxTests.cs  1
         

Complexity decreasing per file
==============================
+ dotnet/SimpleSauce.Test/SauceOptionsTests.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308&bid=15864760)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dotnet/SimpleSauce.Test/Browsers/FirefoxTests.cs  1
- dotnet/Simple.Sauce/SauceSession.cs  1
- dotnet/SimpleSauce.Test/Browsers/ChromeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/EdgeTests.cs  1
- dotnet/SimpleSauce.Test/Browsers/SafariTests.cs  1
         

Complexity decreasing per file
==============================
+ dotnet/SimpleSauce.Test/SauceOptionsTests.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4757308&bid=15864760)

**Pull request merged**